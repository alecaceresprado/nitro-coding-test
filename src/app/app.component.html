<app-header></app-header>
<app-posts-container [posts]="posts$ | async"></app-posts-container>
