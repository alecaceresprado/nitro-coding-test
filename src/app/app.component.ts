import { Component, OnInit } from '@angular/core';
import { Store, select } from '@ngrx/store';
import { Observable } from 'rxjs';

import { Post } from '@models';
import { AppState, fetchPosts, getPosts } from '@state';

@Component({
  selector: 'app-root',
  templateUrl: './app.component.html',
  styleUrls: ['./app.component.scss'],
})
export class AppComponent implements OnInit {
  public posts$: Observable<Post[]>;

  constructor(private store: Store<AppState>) {}

  public ngOnInit() {
    this.store.dispatch(fetchPosts());
    this.posts$ = this.store.pipe(select(getPosts));
  }
}
