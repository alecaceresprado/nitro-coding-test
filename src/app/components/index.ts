export * from './header/header.component';
export * from './posts-container/posts-container.component';
