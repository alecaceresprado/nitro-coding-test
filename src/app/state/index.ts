export * from './post';
export * from './app.reducer';
