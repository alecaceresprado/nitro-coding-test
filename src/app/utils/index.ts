export * from './post.grouper.utils';
