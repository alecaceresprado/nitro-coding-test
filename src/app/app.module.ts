import { BrowserModule } from '@angular/platform-browser';
import { HttpClientModule } from '@angular/common/http';
import { NgModule } from '@angular/core';

import { HeaderComponent } from './components';
import { PostService } from '@services';
import { AppComponent } from './app.component';

@NgModule({
  declarations: [AppComponent, HeaderComponent],
  imports: [BrowserModule, HttpClientModule],
  providers: [PostService],
  bootstrap: [AppComponent],
})
export class AppModule {}
