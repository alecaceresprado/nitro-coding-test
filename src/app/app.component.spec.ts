import { NO_ERRORS_SCHEMA } from '@angular/core';
import { TestBed, async, ComponentFixture } from '@angular/core/testing';
import { provideMockActions } from '@ngrx/effects/testing';
import { provideMockStore } from '@ngrx/store/testing';

import { Post } from '@models';
import { AppComponent } from './app.component';

describe('AppComponent', () => {
  let component: AppComponent;
  let fixture: ComponentFixture<AppComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [],
      declarations: [AppComponent],
      schemas: [NO_ERRORS_SCHEMA],
      providers: [
        provideMockStore({ initialState: { posts: [] } }),
        provideMockActions({} as any),
      ],
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(AppComponent);
    component = fixture.componentInstance;
  });

  it('should create the app', () => {
    expect(fixture).toBeTruthy();
  });

  it('should render', () => {
    expect(fixture).toMatchSnapshot();
  });

  it('should fetch posts - initial state', () => {
    component.ngOnInit();
    component.posts$.subscribe((posts: Post[]) => {
      expect(posts).toMatchSnapshot();
    });
  });
});
