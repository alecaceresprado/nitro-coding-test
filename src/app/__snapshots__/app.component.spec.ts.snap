// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent should fetch posts - initial state 1`] = `Array []`;

exports[`AppComponent should render 1`] = `
<app-root
  store={[Function MockStore]}
>
  <app-header /><app-posts-container />
</app-root>
`;
